### ISSUE TITLE
build_ascent_hip_frontier.sh fails when compiling vtkm

### ISSUE THREAD

**Issue by mlohry** [Alpine-DAV/ascent#1415]
Using `build_ascent_hip_frontier.sh` on frontier, it looks like some part of vtkm isn't properly including mpi:

```
[...]/ascent/scripts/build_ascent/source/vtk-m-v2.1.0/vtkm/filter/flow/internal/BoundsMap.h:25:10: fatal error: 'mpi.h' file not found
```

If I force the include with `export CPATH=/opt/cray/pe/mpich/8.1.28/ofi/crayclang/17.0/include/` it passes this point.

**Comment by cyrush** [comment]
When compiling with the cray compilers, they know where MPI exists.

However, in VTK-m, the compile drops to only use amdclang++ to compile device code and it does not know about implicit mpi paths.

The fix is to add the MPI include path to general includes, we can add this to the frontier script.

**Comment by cyrush** [comment]
@mlohry 

Where you trying the latest script, we do have some logic to address this:

https://github.com/Alpine-DAV/ascent/blob/10b1253c1d0e2ce1a73007b97709f18f7d2b2b07/scripts/build_ascent/build_ascent_hip_frontier.sh#L16

**Comment by mlohry** [comment]
@cyrush  yes I used the latest and I see that line there, odd that it didn't work.

**Comment by BenWibking** [comment]
This appears to be a VTK-m / Viskores bug. I can reproduce this with a local Viskores build on macOS: https://github.com/Viskores/viskores/issues/93.

**Comment by BenWibking** [comment]
This patch should fix the issue: https://github.com/Viskores/viskores/pull/99
